homeassistant/components/climate/knx.py

```diff
@@ -54,11 +64,33 @@
     vol.Optional(CONF_OPERATION_MODE_STATE_ADDRESS): cv.string,
     vol.Optional(CONF_CONTROLLER_STATUS_ADDRESS): cv.string,
     vol.Optional(CONF_CONTROLLER_STATUS_STATE_ADDRESS): cv.string,
+    vol.Optional(CONF_CONTROLLER_MODE_ADDRESS): cv.string,
+    vol.Optional(CONF_CONTROLLER_MODE_STATE_ADDRESS): cv.string,
     vol.Optional(CONF_OPERATION_MODE_FROST_PROTECTION_ADDRESS): cv.string,
     vol.Optional(CONF_OPERATION_MODE_NIGHT_ADDRESS): cv.string,
     vol.Optional(CONF_OPERATION_MODE_COMFORT_ADDRESS): cv.string,
+    vol.Optional(CONF_ON_OFF_ADDRESS): cv.string,
+    vol.Optional(CONF_ON_OFF_STATE_ADDRESS): cv.string,
+    vol.Optional(CONF_OPERATION_MODES): cv.ensure_list_csv,
```

Comment:
Don't use ensure list csv. It's only for backwards compatibility. Use `cv.ensure_list`.

Make sure to validate the values of the list too.

Reply:
Done.